File: InventoryManagement.Core/Entities/Vendor.cs
using InventoryManagement.Core.Common;

namespace InventoryManagement.Core.Entities;

public class Vendor : BaseEntity
{
    public string VendorName { get; set; } = string.Empty;
    public string? Email { get; set; }
    public string? PhoneNumber { get; set; }
    public string? Address { get; set; }
    public string? City { get; set; }
    public string? Country { get; set; }
    public string? PostalCode { get; set; }

    // Navigation Properties
    public ICollection<PurchaseOrder> PurchaseOrders { get; set; } = new List<PurchaseOrder>();
}

File: InventoryManagement.UI/src/app/core/models/vendor.ts
export interface Vendor {
  id: number;
  vendorName: string;
  email?: string;
  phoneNumber?: string;
  address?: string;
  city?: string;
  country?: string;
  postalCode?: string;
  createdAt: Date;
}

export interface CreateVendorRequest {
  vendorName: string;
  email?: string;
  phoneNumber?: string;
  address?: string;
  city?: string;
  country?: string;
  postalCode?: string;
}

export interface UpdateVendorRequest {
  vendorName: string;
  email?: string;
  phoneNumber?: string;
  address?: string;
  city?: string;
  country?: string;
  postalCode?: string;
}